test/query.jl

```diff
@@ -347,6 +347,17 @@ file_path = Path(file_dir)
     end
 end
 
+@testset "Query from IOBuffer" begin
+    streamformat(::Stream{T, U}) where {T, U} = T
+    for name ∈ ["magic1.tiff", "magic2.tiff"]
+        pth = joinpath(file_dir, "magic2.tiff")
+        open(pth) do io
+            buf=IOBuffer(read(io))
+            @test streamformat(query(buf)) == DataFormat{:TIFF}
```

Comment:
How do you know it's TIFF and not OMETIFF? AFAICT only the name distinguishes them.

Reply:
Sorry, now I see, you default to TIFF unless it can prove to itself that it's an OMETIFF via `name`. :+1:

Reply:
I could also try looking for an XML formatted data block, but adding enough of a TIFF parser to find that didn't seem worthwhile.

Reply:
Yeah, technically it's only the name that distinguishes the two file formats and the presence of an OME-XML tag inside the TIFF, but that's difficult to probe for in a general package like this one.